Added gif download modal

holykol opened the pull request:
Fixes #436

The gif should now appear inside the modal instead of appearing at the bottom of the page.
![](https://user-images.githubusercontent.com/22733656/47454537-234da900-d7d8-11e8-9d5d-1552e82777d8.gif)

Please let me know if i'm doing something wrong.

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

jywarren commented on examples/demo.css:
```diff
@@ -132,8 +132,8 @@ h1 {
-  padding: 50px;
   margin: 0 auto;
-
+  width: 100%;
```
Hi, this is great, but actually can we remove the `height` style so that the image is displayed at the correct aspect ratio? Thank you!!! 👍

holykol replied:
@jywarren i'm afraid this is not css fault. The gif gets generated with static width and height. See [gifshot.createGIF options](https://github.com/publiclab/image-sequencer/pull/442/files/c53c730c9cf263aa78b96e7fcb504a2d5fc1e8f1#diff-5b0e3367cfabe5a0173e80753b827d6cR134)

jywarren commented on examples/index.html:
```diff
@@ -103,15 +121,15 @@ <h1>Image Sequencer</h1>
     })
 
+
     function create_gif(){
```
I also wonder if the gif-related code could be moved into a file like some of our other UI code:

```html
   <script src="lib/defaultHtmlStepUi.js" charset="utf-8"></script>
```

It could be in `lib/gif.js` perhaps? That would help keep the demo code tidier!

rockstar777 commented:
Great Work!
Also can you rename "download as gif" to "View Gif" and add download button as "download" itself in popup box?
Thanks for the PR!

holykol commented:
@rockstar777 @jywarren Here we go. ![](https://user-images.githubusercontent.com/22733656/47569636-ca4b5580-d93c-11e8-98c9-147e1063880e.gif)

I have added [download.js](http://danml.com/download.html) to be able to download large files.

Also i have fixed gif aspect ratio. Now it's should use width and height from previous thumbnails.

jywarren commented:
This is great. Since the default download is PNG, could we change Download
to "Download PNG" and then this will be ready? I know it's not in the
original scope but it'll help clarify this now that the Gif button is
there. Thank you!

On Fri, Oct 26, 2018 at 10:00 AM Timofey Barinov <notifications@github.com>
wrote:

> @rockstar777 <https://github.com/rockstar777> @jywarren
> <https://github.com/jywarren> Here we go.
> <https://user-images.githubusercontent.com/22733656/47569636-ca4b5580-d93c-11e8-98c9-147e1063880e.gif>
>
> I have added download.js <http://danml.com/download.html> to be able to
> download large files.
>
> Also i have fixed gif aspect ratio. Now it's should use width and height
> from previous thumbnails.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/image-sequencer/pull/442#issuecomment-433417469>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ4Qq4dACqKrIPXblXv1lplC12AbPks5uoxVNgaJpZM4X4nDl>
> .
>

holykol commented:
@jywarren glad to help you. I think this can finally be merged.

jywarren commented:
This looks awesome. Thanks so much!!!

jywarren commented:
We'd love your help with another project! Perhaps #432 if you are part of GCI, or if not, you might be able to create the UI for that since the GCI folks will be focused on choosing the right icons. What do you think? 

Thanks!!!

rockstar777 commented:
you made it awesome!
Thank you @holykol
